agent/src/beerocks/slave/beerocks_slave_main.cpp

```diff
@@ -186,21 +186,25 @@ static void son_slave_watchdog(beerocks::config_file::sConfigSlave &beerocks_sla
                                const std::unordered_map<int, std::string> &interfaces_map)
 {
     for (int slave_num = 0; slave_num < beerocks::IRE_MAX_SLAVES; slave_num++) {
-        if (!beerocks_slave_conf.hostap_iface[slave_num].empty()) {
-            // check if slave is still running
-            std::string base_name =
-                std::string(BEEROCKS_AGENT) + "_" + beerocks_slave_conf.hostap_iface[slave_num];
-            if (!beerocks::os_utils::is_pid_running(beerocks_slave_conf.temp_path, base_name)) {
-                //start new slave process
-                std::string file_name = "./" + std::string(BEEROCKS_AGENT);
-                if (access(file_name.c_str(), F_OK) == -1) //file does not exist in current location
-                {
-                    file_name = BEEROCKS_BIN_PATH + std::string(BEEROCKS_AGENT);
+        auto hostap_iface_elm = interfaces_map.find(slave_num);
+        if (hostap_iface_elm != interfaces_map.end()) {
```

Comment:
Why not return in case the interface is not found and prevent one level of indentation?
```suggestion
        if (hostap_iface_elm == interfaces_map.end()) {
            return;
        }
        ...
```

Reply:
suggestion is taken. changed to a negative check. if the negative condition is met - continue.